### ISSUE TITLE
WHERE to set the EXECUTOR_ID in  set "useExecutor" = EXECUTOR_ID

### ISSUE THREAD

**Issue by JayMiao** [azkaban/azkaban#667]
I wanna "Force execution to an executor" , But  where should I set the EXECUTOR_ID ?
In which file and in what style ?

**Comment by devangorder** [comment]
I have only been able to successfully force execution to an executor by setting the 'useExecutor' flow override within the web interface when launching a job. Ideally, I'd like to be able to set this within the job files themselves when loading the project (see #656) but it doesn't seem to respect that configuration.

**Comment by logiclord** [comment]
@JayMiao 

@devangorder  you are right setting it within UI or setting it using http request are the only two ways. Only job properties can be set in job files and 'useExecutor' is a flow property.

**Comment by devangorder** [comment]
@logiclord is it possible to set useExecutor in the http request when scheduling with the API calls? can you provide an example if so?

**Comment by logiclord** [comment]
curl 'https://localhost:8443/executor?projectId=1&project=acceptance-test&ajax=executeFlow&flow=all-acceptance-tests&disabled=%5B%5D&failureEmailsOverride=false&successEmailsOverride=false&failureAction=finishCurrent&failureEmails=azktest%40linkedin.com&successEmails=azktest%40linkedin.com&notifyFailureFirst=false&notifyFailureLast=false&flowOverride%5B**useExecutor**%5D=1&concurrentOption=skip' -H 'Cookie: bcookie="v=blah-blah-blah-blah-blah-blah-blah"

**Comment by JayMiao** [comment]
@logiclord 
But where could I get the value of EXECUTOR_ID for a 'useExecutor' ?

**Comment by logiclord** [comment]
from executors table in mysql/h2

**Comment by tk0485** [comment]
I'm trying to set the executor while scheduling the jobs. I added "useExecutor" to the curl command below for scheduling jobs but it doesn't seem to do anything? 

`curl -k https://localhost:8443/schedule -d  "ajax=scheduleFlow&projectNme=testProject&flow=flowName&projectId=1&useExecutor=1&scheduleTime=7,00,am,PDT&scheduleDate=07/20/2016"  -b azkaban.browser.session.id=6c96e7d8-4df5-470d-88fe-259392c09eea`

Also, I'm wondering if I add seconds to the time for scheduling will have any impact on the scheduling time?

**Comment by logiclord** [comment]
You are not passing useExecutor correctly. It should be passed as flow params like:-
curl 'https://localhost:8443/executor?projectId=1&project=acceptance-test&ajax=executeFlow&flow=all-acceptance-tests&disabled=%5B%5D&failureEmailsOverride=false&successEmailsOverride=false&failureAction=finishCurrent&failureEmails=azktest%40linkedin.com&successEmails=azktest%40linkedin.com&notifyFailureFirst=false&notifyFailureLast=false&flowOverride%5BuseExecutor%5D=1&concurrentOption=skip' -H 'Cookie: bcookie="v=blah-blah-blah-blah-blah-blah-blah"

**Comment by vermamitesh87** [comment]
@logiclord I read previous comments that EXECUTOR_ID  is a flow parameter. But how can I force azkaban to run a job from a particular executor?

**Comment by tk0485** [comment]
This issue is resolved I can set executor using through the api like this:

`data = {'ajax':'scheduleFlow','projectName':projectName,'flow': flow,'projectId':str(projectId), 'scheduleTime':time,'scheduleDate':date, 'session.id': sessionId, "flowOverride[useExecutor]": executorID}`
